jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -963,6 +971,21 @@ private static final MethodHandle resolveInvokeDynamic(Class<?> clazz, String na
 		return result;
 	}
 	
+	/*
+	 * Throw NoClassDefFoundError if the cause of TypeNotPresentException is 
+	 * ClassNotFoundException. Otherwise, re-throw TypeNotPresentException.
+	 */
+	private static void throwNoClassDefFoundError(TypeNotPresentException e) {
```

Comment:
And this should be declared as:
```
private static Throwable throwNoClassDefFoundError(TypeNotPresentException e)  {
```
Even though its declared to return the Throwable, it should still throw the exception itself.  This is just a pattern to make it easier for the caller to make it explicit that there is no fall through case

Reply:
fixed.